src/result_management/read_results.py

```diff
@@ -64,3 +65,212 @@ def extract_dataset_from_h5(dataset: h5py.Dataset) -> list:
     data = [item.decode("utf-8") for item in dataset]
 
     return data
+
+
+def add_values_to_summary(summary_path: Path or str, component_set: list = None):
+    """
+    Collect values of input cost parameters and relevant variables from HDF5 files and add them to the summary Excel file.
+
+    Args:
+        summary_path (Path or str): Path to the summary Excel file.
+        component_set (list, optional): List of components to extract parameters and variables from.
+            Defaults to ["Technologies", "Networks", "Import", "Export"].
+    """
+
+    if component_set is None:
+        component_set = {"Technologies", "Networks", "Import", "Export"}
+
+    # Ensure summary_path is a Path object
+    if not isinstance(summary_path, Path):
+        summary_path = Path(summary_path)
+
+    summary_results = pd.read_excel(summary_path)
+
+    # paths to results
+    paths = {}
+    for timestamp in summary_results["time_stamp"].unique():
+        paths[timestamp] = list(
+            summary_results.loc[
+                summary_results["time_stamp"] == timestamp, "time_stamp"
+            ].values
+        )
+
+    # dicts to store data
+    output_dict = {}
+
+    # Extract data from h5 files
+
+    for case in paths:
+        path = Path(case)
+        hdf_file_path = path / "optimization_results.h5"
+        if hdf_file_path.exists():
+            with h5py.File(hdf_file_path, "r") as hdf_file:
+
+                if "Technologies" in component_set:
+                    df = extract_datasets_from_h5group(hdf_file["design/nodes"]).sum()
+                    for period in df.index.levels[0]:
+                        for node in df.index.levels[1]:
+                            for tec in df.index.levels[2]:
+                                parameters = [
+                                    "size",
+                                    "capex_tot",
+                                    "para_unitCAPEX",
+                                    "para_fixCAPEX",
+                                ]
+                                for para in parameters:
+                                    output_name = f"{period}/{node}/{tec}/{para}"
+                                    try:
```

Comment:
Can we write this try statement more specifically? Or without a try statement? Like:

if these columns are in the df:
do blaa

Otherwise it skips the statement for whatever error it gets there (bad practice, sorry for being picky :D)

Reply:
done